tests/Tests.FeatureManagement/RecurrenceEvaluation.cs

```diff
@@ -0,0 +1,1740 @@
+                            Interval = 2,
+                            // FirstDayOfWeek is Sunday by default.
+                            DaysOfWeek = new List<DayOfWeek>(){ DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday, DayOfWeek.Saturday, DayOfWeek.Sunday }
+                        },
+                        Range = new RecurrenceRange()
+                        {
+                            Type = RecurrenceRangeType.Numbered,
+                            NumberOfOccurrences = 3
+                        }
+                    }
+                },
+                null),
+
+                ( DateTimeOffset.Parse("2024-2-11T00:00:00+08:00"), // Sunday in the 2nd week
+                new TimeWindowFilterSettings()
+                {
+                    Start = DateTimeOffset.Parse("2024-2-1T12:00:00+08:00"), // Thursday
+                    End = DateTimeOffset.Parse("2024-2-1T12:00:01+08:00"),
+                    Recurrence = new Recurrence()
+                    {
+                        Pattern = new RecurrencePattern()
+                        {
+                            Type = RecurrencePatternType.Weekly,
+                            Interval = 2,
+                            FirstDayOfWeek = DayOfWeek.Monday,
+                            DaysOfWeek = new List<DayOfWeek>(){ DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday, DayOfWeek.Saturday, DayOfWeek.Sunday }
+                        },
+                        Range = new RecurrenceRange()
+                        {
+                            Type = RecurrenceRangeType.Numbered,
+                            NumberOfOccurrences = 3
+                        }
+                    }
+                },
+                null),
+
+                ( DateTimeOffset.Parse("2024-2-11T00:00:00+08:00"),
+                new TimeWindowFilterSettings()
+                {
+                    Start = DateTimeOffset.Parse("2024-2-1T00:00:00+08:00"), // Thursday
+                    End = DateTimeOffset.Parse("2024-2-1T00:00:01+08:00"),
+                    Recurrence = new Recurrence()
+                    {
+                        Pattern = new RecurrencePattern()
+                        {
+                            Type = RecurrencePatternType.Weekly,
+                            DaysOfWeek = new List<DayOfWeek>(){ DayOfWeek.Thursday, DayOfWeek.Sunday }
+                        },
+                        Range = new RecurrenceRange()
+                        {
+                            Type = RecurrenceRangeType.EndDate,
+                            EndDate = DateTimeOffset.Parse("2024-2-8T00:00:00+08:00")
+                        }
+                    }
+                },
+                null)
+            };
+
+            ConsumeEvalutationTestData(testData);
+        }
+
+        [Fact]
+        public async void RecurrenceEvaluationThroughCacheTest()
```

Comment:
Shall we check if there's an entry in cache?

Reply:
I just sent a PR #445 to test it.